src/UI/Components/ButtonS/ButtonS.tsx

```diff
@@ -2,14 +2,17 @@ import React from 'react'
 import ButtonStyled from './ButtonStyled'
 
 type Props = {
-    onClick?: () => void
+    name: string
+    onClick: (name: string) => void
     children?: string
-    showIcon?: boolean | undefined
+    showIcon?: boolean
 }
-const ButtonS: React.FC<Props> = ({ children, showIcon, onClick }) => {
+const ButtonS: React.FC<Props> = ({ children, showIcon, onClick, name }) => {
+    const setFilter = () => onClick(name)
+
     return (
-        <ButtonStyled onClick={onClick} showIcon={showIcon}>
-            {showIcon ? <div className="icon" /> : null}
+        <ButtonStyled onClick={setFilter} showIcon={showIcon}>
+            {showIcon ? <div className="icon" role={'img'} /> : null}
```

Comment:
Instead of rendering `null`, you can only attend the case in which the condition returns `true` using `&&`. For example:

`{showIcon && <div />}`

See: [&& operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND)